src/python/pants/backend/typescript/dependency_inference/rules.py

```diff
@@ -112,7 +107,11 @@ async def infer_typescript_source_dependencies(
     sources = await Get(
         HydratedSources, HydrateSourcesRequest(source, for_sources_types=[TypeScriptSourceField])
     )
-    metadata = await _prepare_inference_metadata(request.field_set.address, source.file_path)
+    owning_pkg, maybe_config = await concurrently(
+        find_owning_package(OwningNodePackageRequest(request.field_set.address)),
+        find_parent_ts_config(ParentTSConfigRequest(source.file_path, "tsconfig.json"), **implicitly()),
+    )
+    metadata = await _prepare_inference_metadata(owning_pkg, maybe_config, request.field_set.address.spec_path)
 
     import_strings, candidate_pkgs = await concurrently(
         parse_javascript_deps(NativeDependenciesRequest(sources.snapshot.digest, metadata)),
```

Comment:
Word of warning that this parses javascript and JSX files, and will fail to parse as soon as it encounters the `type` keyword, which is valid for typescript but not JS.